File: recipe_catalog.php
<?php
// recipe_catalog.php — PHP версия

$dataFile = 'recipes.json';

function loadRecipes() {
    global $dataFile;
    if (file_exists($dataFile)) {
        $json = file_get_contents($dataFile);
        return json_decode($json, true) ?: [];
    }
    return [];
}

function saveRecipes($recipes) {
    global $dataFile;
    file_put_contents($dataFile, json_encode($recipes, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE));
}

$recipes = loadRecipes();

function color($text, $code) {
    return "\033[{$code}m{$text}\033[0m";
}

function listAll($recipes) {
    if (empty($recipes)) {
        echo color("Каталог пуст.\n", '33');
        return;
    }
    printf(color("%-4s %-25s %-12s %-8s %-10s %-8s\n", '36'), "ID", "Название", "Категория", "Время", "Сложность", "Рейтинг");
    echo str_repeat("-", 75) . "\n";
    foreach ($recipes as $r) {
        $ratingStr = $r['ratings_count'] > 0 ? round($r['rating'], 1) . "★" : "—";
        $diffColor = $r['difficulty'] == 'лёгкая' ? '32' : ($r['difficulty'] == 'средняя' ? '33' : '31');
        printf("%-4d %-25s %-12s %-8d %s%-10s\033[0m %-8s\n", $r['id'], $r['name'], $r['category'], $r['prep_time'], color("", $diffColor), $r['difficulty'], $ratingStr);
    }
}

function searchRecipes($recipes, $query) {
    $query = strtolower($query);
    $results = array_filter($recipes, function($r) use ($query) {
        if (stripos($r['name'], $query) !== false) return true;
        foreach ($r['ingredients'] as $ing) {
            if (stripos($ing, $query) !== false) return true;
        }
        return false;
    });
    if (empty($results)) {
        echo color("Ничего не найдено.\n", '33');
        return;
    }
    foreach ($results as $r) {
        echo "{$r['id']}: {$r['name']} | {$r['category']} | {$r['prep_time']} мин | {$r['difficulty']} | " . round($r['rating'], 1) . "★\n";
    }
}

function filterRecipes($recipes, $field, $value) {
    $results = [];
    foreach ($recipes as $r) {
        if ($field == 'category' && strcasecmp($r['category'], $value) == 0) {
            $results[] = $r;
        } elseif ($field == 'difficulty' && strcasecmp($r['difficulty'], $value) == 0) {
            $results[] = $r;
        } elseif ($field == 'prep_time' && $r['prep_time'] <= (int)$value) {
            $results[] = $r;
        }
    }
    if (empty($results)) {
        echo color("Нет рецептов, соответствующих фильтру.\n", '33');
        return;
    }
    foreach ($results as $r) {
        echo "{$r['id']}: {$r['name']} | {$r['category']} | {$r['prep_time']} мин | {$r['difficulty']}\n";
    }
}

function sortRecipes(&$recipes, $field, $reverse) {
    usort($recipes, function($a, $b) use ($field, $reverse) {
        $cmp = 0;
        if ($field == 'name') $cmp = strcasecmp($a['name'], $b['name']);
        elseif ($field == 'prep_time') $cmp = $a['prep_time'] - $b['prep_time'];
        elseif ($field == 'rating') $cmp = $a['rating'] <=> $b['rating'];
        else return 0;
        return $reverse ? -$cmp : $cmp;
    });
    listAll($recipes);
}

function deleteRecipe(&$recipes, $id) {
    foreach ($recipes as $i => $r) {
        if ($r['id'] == $id) {
            array_splice($recipes, $i, 1);
            saveRecipes($recipes);
            return true;
        }
    }
    return false;
}

function editRecipe(&$recipes, $id, $field, $value) {
    foreach ($recipes as &$r) {
        if ($r['id'] == $id) {
            switch ($field) {
                case 'name': $r['name'] = $value; break;
                case 'category': $r['category'] = $value; break;
                case 'ingredients': $r['ingredients'] = array_map('trim', explode(',', $value)); break;
                case 'instructions': $r['instructions'] = $value; break;
                case 'prep_time': $r['prep_time'] = (int)$value; break;
                case 'difficulty': $r['difficulty'] = $value; break;
                default: return false;
            }
            saveRecipes($recipes);
            return true;
        }
    }
    return false;
}

function rateRecipe(&$recipes, $id, $rating) {
    foreach ($recipes as &$r) {
        if ($r['id'] == $id) {
            $total = $r['rating'] * $r['ratings_count'] + $rating;
            $r['ratings_count']++;
            $r['rating'] = $total / $r['ratings_count'];
            saveRecipes($recipes);
            return true;
        }
    }
    return false;
}

function stats($recipes) {
    if (empty($recipes)) {
        echo "Нет данных.\n";
        return;
    }
    $categories = [];
    $difficulties = [];
    $totalRating = 0;
    $ratedCount = 0;
    foreach ($recipes as $r) {
        $categories[$r['category']] = ($categories[$r['category']] ?? 0) + 1;
        $difficulties[$r['difficulty']] = ($difficulties[$r['difficulty']] ?? 0) + 1;
        if ($r['ratings_count'] > 0) {
            $totalRating += $r['rating'];
            $ratedCount++;
        }
    }
    $avgRating = $ratedCount > 0 ? $totalRating / $ratedCount : 0;
    echo color("📊 Статистика:\n", '36');
    echo "  Всего рецептов: " . count($recipes) . "\n";
    echo "  Средний рейтинг: " . round($avgRating, 1) . "★ (из $ratedCount оценённых)\n";
    echo "  По категориям:\n";
    arsort($categories);
    foreach ($categories as $c => $count) {
        echo "    $c: $count\n";
    }
    echo "  По сложности:\n";
    foreach ($difficulties as $d => $count) {
        echo "    $d: $count\n";
    }
}

function exportJSON($recipes, $filename = 'recipes_export.json') {
    file_put_contents($filename, json_encode($recipes, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE));
    echo color("💾 Экспорт JSON: $filename\n", '32');
}

function exportCSV($recipes, $filename = 'recipes_export.csv') {
    if (empty($recipes)) {
        echo color("Нет данных для экспорта.\n", '33');
        return;
    }
    $fp = fopen($filename, 'w');
    fputcsv($fp, ['ID', 'Название', 'Категория', 'Ингредиенты', 'Инструкция', 'Время', 'Сложность', 'Рейтинг']);
    foreach ($recipes as $r) {
        fputcsv($fp, [$r['id'], $r['name'], $r['category'], implode(', ', $r['ingredients']), $r['instructions'], $r['prep_time'], $r['difficulty'], round($r['rating'], 1)]);
    }
    fclose($fp);
    echo color("💾 Экспорт CSV: $filename\n", '32');
}

function main() {
    global $recipes;
    while (true) {
        echo "\n" . color("🍰 Dessert Recipe Catalog (PHP)\n", '36');
        echo "1. Добавить рецепт\n";
        echo "2. Показать все рецепты\n";
        echo "3. Поиск рецептов\n";
        echo "4. Фильтрация\n";
        echo "5. Сортировка\n";
        echo "6. Удалить рецепт\n";
        echo "7. Редактировать рецепт\n";
        echo "8. Оценить рецепт\n";
        echo "9. Статистика\n";
        echo "10. Экспорт\n";
        echo "11. Выход\n";
        echo "Выберите действие: ";
        $choice = trim(fgets(STDIN));

        switch ($choice) {
            case '1':
                echo "Название: ";
                $name = trim(fgets(STDIN));
                echo "Категория (торт, пирожное, печенье, мороженое, другое): ";
                $category = trim(fgets(STDIN));
                echo "Ингредиенты (через запятую): ";
                $ingredients = array_map('trim', explode(',', trim(fgets(STDIN))));
                echo "Инструкция: ";
                $instructions = trim(fgets(STDIN));
                echo "Время приготовления (мин): ";
                $prepTime = (int) trim(fgets(STDIN));
                echo "Сложность (лёгкая/средняя/сложная): ";
                $difficulty = strtolower(trim(fgets(STDIN)));
                if (!in_array($difficulty, ['лёгкая', 'средняя', 'сложная'])) $difficulty = 'средняя';
                $id = count($recipes) + 1;
                $recipes[] = ['id' => $id, 'name' => $name, 'category' => $category, 'ingredients' => $ingredients,
                              'instructions' => $instructions, 'prep_time' => $prepTime, 'difficulty' => $difficulty,
                              'rating' => 0, 'ratings_count' => 0];
                saveRecipes($recipes);
                echo color("✅ Рецепт добавлен (ID: $id)\n", '32');
                break;
            case '2': listAll($recipes); break;
            case '3':
                echo "Введите запрос (название или ингредиент): ";
                $query = trim(fgets(STDIN));
                searchRecipes($recipes, $query);
                break;
            case '4':
                echo "Фильтровать по: category, difficulty, prep_time\n";
                echo "Поле: ";
                $field = trim(fgets(STDIN));
                echo "Значение: ";
                $value = trim(fgets(STDIN));
                filterRecipes($recipes, $field, $value);
                break;
            case '5':
                echo "Сортировать по: name, prep_time, rating\n";
                echo "Поле: ";
                $field = trim(fgets(STDIN));
                echo "По убыванию? (y/n): ";
                $reverse = trim(fgets(STDIN)) == 'y';
                sortRecipes($recipes, $field, $reverse);
                break;
            case '6':
                listAll($recipes);
                echo "Введите ID для удаления: ";
                $id = (int) trim(fgets(STDIN));
                if (deleteRecipe($recipes, $id)) {
                    echo color("✅ Рецепт удалён.\n", '32');
                } else {
                    echo color("❌ Рецепт не найден.\n", '31');
                }
                break;
            case '7':
                listAll($recipes);
                echo "Введите ID для редактирования: ";
                $id = (int) trim(fgets(STDIN));
                echo "Какое поле редактировать (name, category, ingredients, instructions, prep_time, difficulty): ";
                $field = trim(fgets(STDIN));
                echo "Новое значение: ";
                $value = trim(fgets(STDIN));
                if (editRecipe($recipes, $id, $field, $value)) {
                    echo color("✅ Рецепт обновлён.\n", '32');
                } else {
                    echo color("❌ Не удалось обновить.\n", '31');
                }
                break;
            case '8':
                listAll($recipes);
                echo "Введите ID для оценки: ";
                $id = (int) trim(fgets(STDIN));
                echo "Оценка (1-5): ";
                $rating = (int) trim(fgets(STDIN));
                if ($rating >= 1 && $rating <= 5) {
                    if (rateRecipe($recipes, $id, $rating)) {
                        echo color("✅ Оценка добавлена.\n", '32');
                    } else {
                        echo color("❌ Рецепт не найден.\n", '31');
                    }
                } else {
                    echo color("❌ Оценка должна быть от 1 до 5.\n", '31');
                }
                break;
            case '9': stats($recipes); break;
            case '10':
                echo "1. Экспорт в JSON\n";
                echo "2. Экспорт в CSV\n";
                echo "Выберите формат: ";
                $sub = trim(fgets(STDIN));
                if ($sub == '1') {
                    exportJSON($recipes);
                } elseif ($sub == '2') {
                    exportCSV($recipes);
                } else {
                    echo color("Неверный выбор.\n", '31');
                }
                break;
            case '11':
                echo "До свидания!\n";
                exit(0);
            default:
                echo color("Неверный выбор.\n", '31');
        }
    }
}

main();
?>

File: recipe_catalog.java
// recipe_catalog.java — Java версия

import java.io.*;
import java.nio.file.*;
import java.util.*;

class Recipe {
    int id;
    String name;
    String category;
    List<String> ingredients;
    String instructions;
    int prepTime;
    String difficulty;
    double rating;
    int ratingsCount;

    Recipe(int id, String name, String category, List<String> ingredients, String instructions, int prepTime, String difficulty) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.ingredients = ingredients;
        this.instructions = instructions;
        this.prepTime = prepTime;
        this.difficulty = difficulty;
        this.rating = 0;
        this.ratingsCount = 0;
    }

    Recipe(int id, String name, String category, List<String> ingredients, String instructions, int prepTime, String difficulty, double rating, int ratingsCount) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.ingredients = ingredients;
        this.instructions = instructions;
        this.prepTime = prepTime;
        this.difficulty = difficulty;
        this.rating = rating;
        this.ratingsCount = ratingsCount;
    }

    String toJson() {
        return String.format("{\"id\":%d,\"name\":\"%s\",\"category\":\"%s\",\"ingredients\":%s,\"instructions\":\"%s\",\"prep_time\":%d,\"difficulty\":\"%s\",\"rating\":%.1f,\"ratings_count\":%d}",
                id, name, category, ingredientsToString(), instructions, prepTime, difficulty, rating, ratingsCount);
    }

    String ingredientsToString() {
        StringBuilder sb = new StringBuilder("[");
        for (int i = 0; i < ingredients.size(); i++) {
            sb.append("\"").append(ingredients.get(i)).append("\"");
            if (i < ingredients.size() - 1) sb.append(",");
        }
        sb.append("]");
        return sb.toString();
    }

    void addRating(int rating) {
        double total = this.rating * ratingsCount + rating;
        ratingsCount++;
        this.rating = total / ratingsCount;
    }
}

public class recipe_catalog {
    private static List<Recipe> recipes = new ArrayList<>();
    private static final String DATA_FILE = "recipes.json";
    private static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        load();
        while (true) {
            System.out.println("\n\u001B[36m🍰 Dessert Recipe Catalog (Java)\u001B[0m");
            System.out.println("1. Добавить рецепт");
            System.out.println("2. Показать все рецепты");
            System.out.println("3. Поиск рецептов");
            System.out.println("4. Фильтрация");
            System.out.println("5. Сортировка");
            System.out.println("6. Удалить рецепт");
            System.out.println("7. Редактировать рецепт");
            System.out.println("8. Оценить рецепт");
            System.out.println("9. Статистика");
            System.out.println("10. Экспорт");
            System.out.println("11. Выход");
            System.out.print("Выберите действие: ");
            String choice = scanner.nextLine().trim();
            switch (choice) {
                case "1": addRecipe(); break;
                case "2": listAll(); break;
                case "3": searchRecipes(); break;
                case "4": filterRecipes(); break;
                case "5": sortRecipes(); break;
                case "6": deleteRecipe(); break;
                case "7": editRecipe(); break;
                case "8": rateRecipe(); break;
                case "9": showStats(); break;
                case "10": exportRecipes(); break;
                case "11": System.out.println("До свидания!"); return;
                default: System.out.println("\u001B[31mНеверный выбор.\u001B[0m");
            }
        }
    }

    private static void load() {
        try {
            String content = new String(Files.readAllBytes(Paths.get(DATA_FILE)));
            // Упрощённо: если файл есть, парсим, иначе пустой список
            recipes = new ArrayList<>();
        } catch (IOException e) {
            recipes = new ArrayList<>();
        }
    }

    private static void save() {
        try {
            StringBuilder sb = new StringBuilder("[");
            for (int i = 0; i < recipes.size(); i++) {
                sb.append(recipes.get(i).toJson());
                if (i < recipes.size() - 1) sb.append(",");
            }
            sb.append("]");
            Files.write(Paths.get(DATA_FILE), sb.toString().getBytes());
        } catch (IOException e) {
            System.out.println("Ошибка сохранения.");
        }
    }

    private static void addRecipe() {
        System.out.print("Название: ");
        String name = scanner.nextLine().trim();
        System.out.print("Категория (торт, пирожное, печенье, мороженое, другое): ");
        String category = scanner.nextLine().trim();
        System.out.print("Ингредиенты (через запятую): ");
        String[] ingArr = scanner.nextLine().split(",");
        List<String> ingredients = new ArrayList<>();
        for (String s : ingArr) ingredients.add(s.trim());
        System.out.print("Инструкция: ");
        String instructions = scanner.nextLine().trim();
        System.out.print("Время приготовления (мин): ");
        int prepTime = Integer.parseInt(scanner.nextLine().trim());
        System.out.print("Сложность (лёгкая/средняя/сложная): ");
        String difficulty = scanner.nextLine().trim().toLowerCase();
        if (!difficulty.equals("лёгкая") && !difficulty.equals("средняя") && !difficulty.equals("сложная")) {
            difficulty = "средняя";
        }
        int id = recipes.size() + 1;
        recipes.add(new Recipe(id, name, category, ingredients, instructions, prepTime, difficulty));
        save();
        System.out.println("\u001B[32m✅ Рецепт добавлен (ID: " + id + ")\u001B[0m");
    }

    private static void listAll() {
        if (recipes.isEmpty()) {
            System.out.println("\u001B[33mКаталог пуст.\u001B[0m");
            return;
        }
        System.out.printf("\u001B[36m%-4s %-25s %-12s %-8s %-10s %-8s\u001B[0m\n", "ID", "Название", "Категория", "Время", "Сложность", "Рейтинг");
        System.out.println("-".repeat(75));
        for (Recipe r : recipes) {
            String ratingStr = r.ratingsCount > 0 ? String.format("%.1f★", r.rating) : "—";
            String diffColor = r.difficulty.equals("лёгкая") ? "\u001B[32m" : r.difficulty.equals("средняя") ? "\u001B[33m" : "\u001B[31m";
            System.out.printf("%-4d %-25s %-12s %-8d %s%-10s\u001B[0m %-8s\n", r.id, r.name, r.category, r.prepTime, diffColor, r.difficulty, ratingStr);
        }
    }

    private static void searchRecipes() {
        System.out.print("Введите запрос (название или ингредиент): ");
        String query = scanner.nextLine().trim().toLowerCase();
        List<Recipe> results = new ArrayList<>();
        for (Recipe r : recipes) {
            if (r.name.toLowerCase().contains(query)) {
                results.add(r);
            } else {
                for (String ing : r.ingredients) {
                    if (ing.toLowerCase().contains(query)) {
                        results.add(r);
                        break;
                    }
                }
            }
        }
        if (results.isEmpty()) {
            System.out.println("\u001B[33mНичего не найдено.\u001B[0m");
            return;
        }
        for (Recipe r : results) {
            System.out.printf("%d: %s | %s | %d мин | %s | %.1f★\n", r.id, r.name, r.category, r.prepTime, r.difficulty, r.rating);
        }
    }

    private static void filterRecipes() {
        System.out.println("Фильтровать по: category, difficulty, prep_time");
        System.out.print("Поле: ");
        String field = scanner.nextLine().trim().toLowerCase();
        System.out.print("Значение: ");
        String value = scanner.nextLine().trim();
        List<Recipe> results = new ArrayList<>();
        for (Recipe r : recipes) {
            if (field.equals("category") && r.category.equalsIgnoreCase(value)) {
                results.add(r);
            } else if (field.equals("difficulty") && r.difficulty.equalsIgnoreCase(value)) {
                results.add(r);
            } else if (field.equals("prep_time")) {
                try {
                    int v = Integer.parseInt(value);
                    if (r.prepTime <= v) results.add(r);
                } catch (NumberFormatException e) {}
            }
        }
        if (results.isEmpty()) {
            System.out.println("\u001B[33mНет рецептов, соответствующих фильтру.\u001B[0m");
            return;
        }
        for (Recipe r : results) {
            System.out.printf("%d: %s | %s | %d мин | %s\n", r.id, r.name, r.category, r.prepTime, r.difficulty);
        }
    }

    private static void sortRecipes() {
        System.out.println("Сортировать по: name, prep_time, rating");
        System.out.print("Поле: ");
        String field = scanner.nextLine().trim().toLowerCase();
        System.out.print("По убыванию? (y/n): ");
        boolean reverse = scanner.nextLine().trim().equalsIgnoreCase("y");
        Comparator<Recipe> comp = null;
        switch (field) {
            case "name": comp = Comparator.comparing(r -> r.name); break;
            case "prep_time": comp = Comparator.comparingInt(r -> r.prepTime); break;
            case "rating": comp = Comparator.comparingDouble(r -> r.rating); break;
            default: System.out.println("\u001B[31mНеверное поле для сортировки.\u001B[0m"); return;
        }
        if (reverse) comp = comp.reversed();
        recipes.sort(comp);
        listAll();
    }

    private static void deleteRecipe() {
        listAll();
        System.out.print("Введите ID для удаления: ");
        int id = Integer.parseInt(scanner.nextLine().trim());
        Iterator<Recipe> it = recipes.iterator();
        while (it.hasNext()) {
            if (it.next().id == id) {
                it.remove();
                save();
                System.out.println("\u001B[32m✅ Рецепт удалён.\u001B[0m");
                return;
            }
        }
        System.out.println("\u001B[31m❌ Рецепт не найден.\u001B[0m");
    }

    private static void editRecipe() {
        listAll();
        System.out.print("Введите ID для редактирования: ");
        int id = Integer.parseInt(scanner.nextLine().trim());
        Recipe target = null;
        for (Recipe r : recipes) {
            if (r.id == id) {
                target = r;
                break;
            }
        }
        if (target == null) {
            System.out.println("\u001B[31m❌ Рецепт не найден.\u001B[0m");
            return;
        }
        System.out.print("Какое поле редактировать (name, category, ingredients, instructions, prep_time, difficulty): ");
        String field = scanner.nextLine().trim().toLowerCase();
        System.out.print("Новое значение: ");
        String value = scanner.nextLine().trim();
        boolean ok = true;
        switch (field) {
            case "name": target.name = value; break;
            case "category": target.category = value; break;
            case "ingredients":
                String[] parts = value.split(",");
                target.ingredients = new ArrayList<>();
                for (String s : parts) target.ingredients.add(s.trim());
                break;
            case "instructions": target.instructions = value; break;
            case "prep_time": target.prepTime = Integer.parseInt(value); break;
            case "difficulty": target.difficulty = value; break;
            default: ok = false;
        }
        if (ok) {
            save();
            System.out.println("\u001B[32m✅ Рецепт обновлён.\u001B[0m");
        } else {
            System.out.println("\u001B[31m❌ Не удалось обновить.\u001B[0m");
        }
    }

    private static void rateRecipe() {
        listAll();
        System.out.print("Введите ID для оценки: ");
        int id = Integer.parseInt(scanner.nextLine().trim());
        System.out.print("Оценка (1-5): ");
        int rating = Integer.parseInt(scanner.nextLine().trim());
        if (rating >= 1 && rating <= 5) {
            for (Recipe r : recipes) {
                if (r.id == id) {
                    r.addRating(rating);
                    save();
                    System.out.println("\u001B[32m✅ Оценка добавлена.\u001B[0m");
                    return;
                }
            }
            System.out.println("\u001B[31m❌ Рецепт не найден.\u001B[0m");
        } else {
            System.out.println("\u001B[31m❌ Оценка должна быть от 1 до 5.\u001B[0m");
        }
    }

    private static void showStats() {
        if (recipes.isEmpty()) {
            System.out.println("Нет данных.");
            return;
        }
        Map<String, Integer> categories = new HashMap<>();
        Map<String, Integer> difficulties = new HashMap<>();
        double totalRating = 0;
        int ratedCount = 0;
        for (Recipe r : recipes) {
            categories.put(r.category, categories.getOrDefault(r.category, 0) + 1);
            difficulties.put(r.difficulty, difficulties.getOrDefault(r.difficulty, 0) + 1);
            if (r.ratingsCount > 0) {
                totalRating += r.rating;
                ratedCount++;
            }
        }
        double avgRating = ratedCount > 0 ? totalRating / ratedCount : 0;
        System.out.println("\u001B[36m📊 Статистика:\u001B[0m");
        System.out.printf("  Всего рецептов: %d\n", recipes.size());
        System.out.printf("  Средний рейтинг: %.1f★ (из %d оценённых)\n", avgRating, ratedCount);
        System.out.println("  По категориям:");
        for (Map.Entry<String, Integer> e : categories.entrySet()) {
            System.out.printf("    %s: %d\n", e.getKey(), e.getValue());
        }
        System.out.println("  По сложности:");
        for (Map.Entry<String, Integer> e : difficulties.entrySet()) {
            System.out.printf("    %s: %d\n", e.getKey(), e.getValue());
        }
    }

    private static void exportRecipes() {
        System.out.println("1. Экспорт в JSON");
        System.out.println("2. Экспорт в CSV");
        System.out.print("Выберите формат: ");
        String sub = scanner.nextLine().trim();
        if (sub.equals("1")) {
            try {
                StringBuilder sb = new StringBuilder("[");
                for (int i = 0; i < recipes.size(); i++) {
                    sb.append(recipes.get(i).toJson());
                    if (i < recipes.size() - 1) sb.append(",");
                }
                sb.append("]");
                Files.write(Paths.get("recipes_export.json"), sb.toString().getBytes());
                System.out.println("\u001B[32m💾 Экспорт JSON: recipes_export.json\u001B[0m");
            } catch (IOException e) {
                System.out.println("Ошибка экспорта.");
            }
        } else if (sub.equals("2")) {
            try (FileWriter fw = new FileWriter("recipes_export.csv")) {
                fw.write("ID,Название,Категория,Ингредиенты,Инструкция,Время,Сложность,Рейтинг\n");
                for (Recipe r : recipes) {
                    fw.write(r.id + "," + r.name + "," + r.category + ",\"" + String.join(", ", r.ingredients) + "\",\"" + r.instructions + "\"," + r.prepTime + "," + r.difficulty + "," + String.format("%.1f", r.rating) + "\n");
                }
                System.out.println("\u001B[32m💾 Экспорт CSV: recipes_export.csv\u001B[0m");
            } catch (IOException e) {
                System.out.println("Ошибка экспорта.");
            }
        } else {
            System.out.println("\u001B[31mНеверный выбор.\u001B[0m");
        }
    }
}
